### PR TITLE
added section to correct corners that are separated by commas instead…

### PR DESCRIPTION
**Pull request by RugnirViking**
… of newlines

**Description**
Encountered an issue in which when upgrading some SimpleBuildings were converted correctly but others appeared flipped. upon investigation it was found that those SimpleBuildings were using comma separated vectors for its corners instead of newline separated like the script expects. This PR goes through and changes any comma-separated lists of corners into newline separated before the file is parsed by the main script

**Additional context**
Persuant to a conversation had in the webots discord on 19th september 2022

### CODE REVIEW

File: scripts/converter/convert_nue_to_enu_rub_to_flu.py [hunk 1]
```diff
@@ -187,6 +187,44 @@ def convert_nue_to_enu_world(filename, mode='all', objects_pi=[], objects_pi_2=[
     rotation_next_object = []
     miss_rotation = False
     last_type = ''
+
+
```
Comment (stefaniapedrazzi) [inline scripts/converter/convert_nue_to_enu_rub_to_flu.py:191]:
```suggestion

```

File: scripts/converter/convert_nue_to_enu_rub_to_flu.py [hunk 2]
```diff
@@ -187,6 +187,44 @@ def convert_nue_to_enu_world(filename, mode='all', objects_pi=[], objects_pi_2=[
...
+        if type in ['corners', 'path', 'wayPoints', 'spine', 'startingAngle', 'endingAngle'] or (
+                        type in ['height'] and 'ElevationGrid' in last_type) or (
+                        type in ['shape'] and 'Crossroad' in last_type):
+            if '[]' not in line:  # if type not empty
+                next_line_is_corners = 1
+            print(line,end='')
+        elif next_line_is_corners == 1:
+            if ']' in line:  # we stop when we reach the end of the node 'point' of 'coord'
+                next_line_is_corners = -1
+                print(line, end ='')
```
Comment (stefaniapedrazzi) [inline scripts/converter/convert_nue_to_enu_rub_to_flu.py:216]:
```suggestion
                print(line, end='')
```

File: scripts/converter/convert_nue_to_enu_rub_to_flu.py [hunk 3]
```diff
@@ -187,6 +187,44 @@ def convert_nue_to_enu_world(filename, mode='all', objects_pi=[], objects_pi_2=[
...
+            else:  # else we convert the line
+                # fix for issue where corners are comma separated
```
Comment (stefaniapedrazzi) [inline scripts/converter/convert_nue_to_enu_rub_to_flu.py:218]:
```suggestion
                # split coordinates separated by commas on new lines
```

File: scripts/converter/convert_nue_to_enu_rub_to_flu.py [hunk 4]
```diff
@@ -187,6 +187,44 @@ def convert_nue_to_enu_world(filename, mode='all', objects_pi=[], objects_pi_2=[
...
+                if "," in line:
+                    vectors = [x for x in re.compile('[,]').split(line)]
+                    for vector in vectors:
+                        print(vector)
```
Comment (stefaniapedrazzi) [inline scripts/converter/convert_nue_to_enu_rub_to_flu.py:222]: The new lines are not correctly aligned.
In this file there is the `add_space` function that is used to compute the spaces needed to correctly align the values.

**Review by stefaniapedrazzi — CHANGES_REQUESTED** [review]
Thank you for this fix!
I tested it and it seems to work fine.

If possible it would be better to avoid duplicating the loop through the file and apply all the changes directly the already existing loop.

**Comment by RugnirViking** [comment]
I have made it so that the vectors are padded correctly, I had a go at putting everything into the single loop but it wasn't working quite right and I need to get back to work @stefaniapedrazzi

File: scripts/converter/convert_nue_to_enu_rub_to_flu.py [hunk 5]
```diff
@@ -187,6 +187,43 @@ def convert_nue_to_enu_world(filename, mode='all', objects_pi=[], objects_pi_2=[
...
+        elif next_line_is_corners == 1:
+            if ']' in line:  # we stop when we reach the end of the node 'point' of 'coord'
+                next_line_is_corners = -1
+                print(line, end='')
+            else:  # else we convert the line
+                # split coordinates separated by commas on new lines
+                if "," in line:
+                    vectors = [x for x in re.compile('[,]').split(line)]
+                    for vector in vectors:
+                        print(add_space(line)+vector.strip())
```
Comment (stefaniapedrazzi) [inline scripts/converter/convert_nue_to_enu_rub_to_flu.py:221]:
```suggestion
                        print(add_space(line) + vector.strip())
```

**Review by stefaniapedrazzi — CHANGES_REQUESTED** [review]
Thank you for fixing the alignment!

There is still just a minor coding style detail to fix then this PR is ready to be merged.

**Comment by omichel** [comment]
@RugnirViking: I just sent you an invite to become a member of the Webots-committers group so that you should be to merge your PR directly and create new PRs on the cyberbotics repos.

**Review by stefaniapedrazzi — APPROVED** [review]
Thank you!

**Pull request merged**